### PR TITLE
PR for #47 zillow rent benchmark

### PR DESCRIPTION
**Pull request by gabrieleborg**
@santiagohermo @diegogentilepassaro 

I have updated the comparison between our preferred Zillow series, SFCC, and the Small Area Fair Market Rent. I am going to show you here several version of this exercise, so that we can quickly discuss which is the most compelling and where we should add this brief data sanity check. All the analysis is performed in [`analysis/zillow_benchmark/code/compare_zillow_zipcode.do`](https://github.com/diegogentilepassaro/min_wage_rent/blob/%2347-zillow-rent-benchmark/analysis/zillow_benchmark/code/compare_zillow_zipcode.do)

First, I have "cleaned" the previous plots showing only our series of interest: 

_time series vs all sfmr (minus 0br series)_
trend_zillow_safmr_zipcode_avg

_time series vs 3br safmr_
trend_zillow_safmr4br_zipcode_avg

_correlation with 3br safmr_
bins_sfcc_safmr3br

Additionally, I have created a weighted safmr series using data from the American Housing Survey (AHS): 
I manually created `drive/raw_data/ahs/AHS_sfcc_est_by_br.csv` (find more info at [`raw/ahs/readme.txt`](https://github.com/diegogentilepassaro/min_wage_rent/blob/%2347-zillow-rent-benchmark/raw/ahs/readme.txt)): The survey provides estimates of the number of housing units by house type and by number of bedrooms. I simply used national-level data so to re-weight the final national averages. The years available are: 2011, 2013, 2015, 2017, 2019. For each year I copied the estimates for those house types that are included by definition in the SFCC series: single families, condos and cooperative houses. This resulted in 4 AHS types: 1family detached, 1family attached, condo, and cooperatives. For each house type, estimates are broken down by number of bedrooms: 0, 1, 2, 3, 4+. Importantly, these estimates were available for rental house and owned house separately, so I only got the former. For each year-bedroom, summed up the number of houses across SFCC types, and created the yearly share. In general, the majority of houses are 2- and 3-br. For the missing years, I assigned values from the previous year (e.g. for 2018 I used 2017 shares). In this way I was able, after computing the national average rent for safmr 0, 1, 2, 3, 4+, to create a weighted average. I then finally plotted the time series of this weighted trend against the yearly national average of our SFCC zillow data: 
trend_zillow_safmrwgt_zipcode_avg

As you can see, we still have a nice comparison showing that our data are reasonable! I was thinking that we could either show this, or the binscatter...or both ? you let me know what you think 😊

### CODE REVIEW

**Review by santiagohermo — COMMENTED** [review]
I finished this review @gabrieleborg. As usual, thanks for the amazing work! Overall the rent benchmarking looks great. I made a few inline comments. General comments below:

- The most important thing I noticed is that you are multiplying by cpi, when I think you should divide. See inline comment.

- Output committed doesn't correspond to output when I ran the folder. Files `trned_zillow_safmr3br` are replaced by `trned_zillow_safmr4br`. Maybe you should commit a new run so that they match.

- My plots have the light blue background. How do you do to have the white background even though you don't specify the options when producing the plots? Should we add the options to produce the same plots when we run it in other machines?

- In the plots labels, shouldn't SAFMR be written in full caps?

File: analysis/zillow_benchmark/code/compare_zillow_zipcode.do [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+	}
+	
+	use zipcode place_code msa countyfips statefips ///
+		year_month `target_vars'           ///
+		using `inzillow'/zipcode_yearmonth_panel_all.dta
+
+		foreach s in `tseries' {
+			g rmtag = (medrentprice_`s'==.)
+			bys zipcode (year_month): g ziplen = _N
+			bys zipcode (year_month): gegen rmcount = sum(rmtag)
```
[the remark this answers is not in the document]
Comment (gabrieleborg) [reply analysis/zillow_benchmark/code/compare_zillow_zipcode.do:68]: yeah, it's egen but faster. Agree

File: analysis/zillow_benchmark/code/compare_zillow_zipcode.do [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+	unab samplist: sample_*
+	gegen allmis = rowtotal(`samplist')
+	drop if allmis==0 
+	drop allmis 
+	g year = yofd(dofm(year_month))
+	keep if inrange(year, 2010, 2019)
+
+	preserve
+	collapse (mean) `target_vars', by(zipcode year)
+	collapse (mean) `target_vars', by(year)
```
[the remark this answers is not in the document]
Comment (gabrieleborg) [reply analysis/zillow_benchmark/code/compare_zillow_zipcode.do:82]: you're right!

File: analysis/zillow_benchmark/code/compare_zillow_zipcode.do [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+	merge 1:1 year using ../temp/wgt_safmr.dta, nogen assert(1 2 3) keep(1 3)
+
+	merge 1:1 year using ../temp/cpi2012.dta, nogen keep(1 3)
+	foreach v in `target_vars' safmr {
+		replace `v' = `v'*cpi2012
+	}
+
+	twoway (line medrentprice_sfcc year, lc(eltblue)) ///
+		   (line safmr year, lp(dash) lc(gs10)), ///
+	legend(order(1 "zillow single family/condo" 2 "safmr")) ///
```
[the remark this answers is not in the document]
Comment (gabrieleborg) [reply analysis/zillow_benchmark/code/compare_zillow_zipcode.do:92]: corrected

**Comment by gabrieleborg** [comment]
Thanks for the review @santiagohermo! 

I updated the code following your suggestions, and I re-run it all so to clean all log files. As for the graph scheme, I am simply using the s1color. If you guys are fine with that simply run `set scheme s1color, perm` once in Stata. 

Thanks for spotting the CPI error!

**Comment by diegogentilepassaro** [comment]
Thanks for the great work @gabrieleborg! This is really looking awesome and it is so useful to know that we are actually measuring rent. I am really looking forward for the next draft as it will be significantly better :D

One quick comment. In the following graph, I don't think that we are showing the correlation. Instead we are showing the coefficient of a linear regression: Cov(y,x)/Var(x). That is very useful to see, but we probably also want the correlation: Cov(y,x)/sqrt(Var(x))sqrt(Var(y))
Screen Shot 2020-10-12 at 10 15 22 AM

Could we add that calculation to the plot so that we can say how much is the time series correlation?

**Comment by gabrieleborg** [comment]
Thanks for the comment @diegogentilepassaro ! 

I added few lines where I compute the correlation coefficient between the two variables, that I then add to the plot together with the estimated slope of the fitted line. 

This however made think a bit more carefully about this, and I have a question: 

1. Right now I am simply plotting the binned relationship between the two measures, and I absorb year FE to reflect the short-term nature of our exercise (i.e. we are interested mostly in short term variation, hence it make sense to compare within given years). However I was not using zipcode FE. When you refer to _time series correlation_ I guess you mean the correlation for the time-series of each zipcode, so I was thinking that maybe it make sense to actually use zipcode FE as opposed to year FE (note that in order to compute the associated correlation rho, I first residualize each variable on the FE used).

**Review by santiagohermo — APPROVED** [review]
Thanks @gabrieleborg! I think the only minor thing that remains open from my comments is [this](https://github.com/diegogentilepassaro/min_wage_rent/pull/52#discussion_r502672581). Other than that I think we can merge

**Comment by diegogentilepassaro** [comment]
Thanks for adding the correlation @gabrieleborg!! As for what is the right correlation, that is actually a great question! My intuition is that the raw time series one would be very useful. In order to compute that, I guess that there are 3 different ways (@santiagohermo check that you agree or if you have other ways!):

1. Collapse each data to the year-month, and that would represent the time-series for the average zipcode in each dataset. 

2. Collapse each data to the year-month using some weights such that both datasets represent the average US zipcode.

3. Collapse each data to the year-zipcode (as the Zillow come in zipcode-year-month and the SAMFR comes in zipcode-year). Regress each series only against zipcode FEs. Save the coefficients, and for each observations of each series subtract its corresponding zipcode FE (basically keep only time variation!). Compute the correlation between the modified series.

File: raw/ahs/readme.txt [hunk 4]
```diff
@@ -0,0 +1,18 @@
+OVERVIEW
+==============================================================
+The dataset contains the estimated count (in thousands) of rental houses for types making up the "single family / Condo Coop" Zillow category, broken down by number of bedrooms in the house. 
+
+Geographic level: national 
+Time level:       years 2011, 2013, 2015, 2017, 2019
+
+SOURCE
+==============================================================
+Data manually compiled from https://www.census.gov/programs-surveys/ahs/data/interactive/ahstablecreator.html?s_areas=00000&s_year=2019&s_tablename=TABLE1&s_bygroup1=1&s_bygroup2=1&s_filtergroup1=1&s_filtergroup2=1
```
[the remark this answers is not in the document]
Comment (gabrieleborg) [reply raw/ahs/readme.txt:10]: I added the options I used in the readme file! I also added a link to the technical documentation for the AHS data.

**Comment by gabrieleborg** [comment]
@diegogentilepassaro at the moment we are doing 3, but with year FE! 

I quickly computed 1 (but only at the year level because SAFMR doesn't have monthly obs), and the correlation is 95%

**Comment by diegogentilepassaro** [comment]
Thanks @gabrieleborg!

If we were doing 3. controlling for time FEs we were essentially using cross sectional (across zipcodes) variation so it makes sense that within year and across zipcodes our Zillow measure correlates only 42% with the SAMFR. Try controlling for zipcode FE only instead and that should give us the time series variation. 

The quick computation from 1. giving 95% confirms the graphical intuition that over time the two series are moving almost identically! We should get something very similar when doing 3. so I left it to you which one you prefer to use!

**Review by diegogentilepassaro — APPROVED** [review]

**Pull request merged**